### PR TITLE
Add tx decoder and utils

### PR DESCRIPTION
**Pull request by kilic**

### CODE REVIEW

File: contracts/libs/Tx.sol [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    uint256 public constant TX_LEN = 12;
+    uint256 public constant MASK_TX = 0xffffffffffffffffffffffff;
+    uint256 public constant MASK_STATE_INDEX = 0xffffffff;
+    uint256 public constant MASK_AMOUNT = 0xffffffff;
+    // positions in bytes
+    uint256 public constant POSITION_SENDER = 4;
+    uint256 public constant POSITION_RECEIVER = 8;
+    uint256 public constant POSITION_AMOUNT = 12;
+
+    function hasExcessData(bytes memory txs) internal pure returns (bool) {
```
Comment (vaibhavchellani) [inline contracts/libs/Tx.sol:65]: Can we have one function here that accepts (uint256 from,uint256 to,uint256 amount,bytes signature) and return bytes that I need to push to `submitBatch` function?

Comment (vaibhavchellani) [reply contracts/libs/Tx.sol:65]: The idea is that the node will call this function with the data and receive the `bytes calldata` that needs to be pushed on-chain.

**Review by vaibhavchellani — APPROVED** [review]
Amazing @kilic! Thanks!

**Comment by ChihChengLiang** [comment]
Does this work only on transfer or on any type of transaction?

**Comment by vaibhavchellani** [comment]
Only on transfers, we will have to do custom stuff for every tx

**Comment by ChihChengLiang** [comment]
Got it, thanks!

**Pull request merged**